service/src/main/java/tds/assessment/repositories/impl/ItemQueryRepositoryImpl.java

```diff
@@ -22,31 +24,96 @@ public ItemQueryRepositoryImpl(final NamedParameterJdbcTemplate jdbcTemplate) {
         this.jdbcTemplate = jdbcTemplate;
     }
 
+    @Override
+    public Map<String, List<Item>> findAllItemsForForms(final List<String> formKeys) {
+        SqlParameterSource parameters = new MapSqlParameterSource("formKeys", formKeys);
+        final String SQL =
+            "SELECT \n" +
+            "   items._key AS id,\n" +
+            "   items.itemtype,\n" +
+            "   formitems._fk_testform AS formKey,\n" +
+            "   admin._fk_adminsubject AS segmentKey,\n" +
+            "   admin.groupid,\n" +
+            "   admin.groupkey,\n" +
+            "   admin.itemposition AS position,\n" +
+            "   admin.isfieldtest,\n" +
+            "   admin.isrequired, \n" +
+            "   admin.strandname \n" +
+            "FROM \n" +
+            "   itembank.tblsetofadminitems AS admin \n" +
+            "JOIN \n" +
+            "   itembank.tblitem items \n" +
+            "   ON items._key = admin._fk_item \n" +
+            "JOIN \n" +
+            "   itembank.testformitem formitems \n" +
+            "   ON formitems._fk_item = items._key \n" +
+            "JOIN " +
+            "   itembank.tblsetofadminsubjects segments \n" +
+            "   ON segments._key = admin._fk_adminsubject \n" +
+            "WHERE \n" +
+            "   (segments.virtualtest = :key " +
+            "       OR segments._key = :key) \n" +
+            "   AND \n" +
+            "       formitems._fk_testform IN (:formKeys) \n" +
+            "   AND \n" +
+            "       formitems.isactive = 1 \n" +
+            "   AND \n" +
+            "       admin.isactive = 1";
```

Comment:
Does this query exist in the legacy application?  If so, what does it do?  I see no issue with the change unless the legacy application makes sure they're the same.  Even if it doesn't necessarily make sense to us that small tweak could introduce a subtle bug.